mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -646,22 +646,28 @@ def _search_runs(self, experiment_ids, filter_string, run_view_type, max_results
                                   INVALID_PARAMETER_VALUE)
 
         stages = set(LifecycleStage.view_type_to_stages(run_view_type))
+
         with self.ManagedSessionMaker() as session:
             # Fetch the appropriate runs and eagerly load their summary metrics, params, and
             # tags. These run attributes are referenced during the invocation of
             # ``run.to_mlflow_entity()``, so eager loading helps avoid additional database queries
             # that are otherwise executed at attribute access time under a lazy loading model.
-            queried_runs = session \
-                .query(SqlRun) \
+            query = session.query(SqlRun)
+
+            parsed = SearchUtils.parse_search_filter(filter_string)
+            for s in _get_sqlalchemy_filter_clauses(parsed, session):
+                query = query.join(s)
+
+            filtered = query.distinct() \
                 .options(*self._get_eager_run_query_options()) \
                 .filter(
                     SqlRun.experiment_id.in_(experiment_ids),
-                    SqlRun.lifecycle_stage.in_(stages)) \
+                    SqlRun.lifecycle_stage.in_(stages),
+                    *_get_attibutes_filtering_clauses(parsed)) \
                 .all()
-            runs = [run.to_mlflow_entity() for run in queried_runs]
```

Comment:
I would have let the variable name 'queried_runs' since it is more representative than 'filtered', we don't always filter and it indicates that the variable contains runs.